client/concordclient/include/client/concordclient/concord_client.hpp

```diff
@@ -130,14 +132,59 @@ class ConcordClient {
 
   // Register a callback that gets invoked for every validated event received.
   // void callback(EventGroup);
+  // Return subscriber ID used to unsubscribe.
   template <class CallbackT>
   void subscribe(const SubscribeRequest& request,
                  const std::unique_ptr<opentracing::Span>& parent_span,
                  CallbackT callback);
+
+  // Unsubscribe with the subscriber ID retrieved from a prior call to subscribe.
+  // Note, if the caller doesn't unsubscribe and no runtime error occurs then resources
+  // will be occupied forever.
   void unsubscribe();
 
  private:
+  logging::Logger logger_;
   ConcordClientConfig config_;
+
+  // TODO: Allow multiple subscriptions
+  std::atomic_bool stop_subscriber_{true};
+  std::thread subscriber_;
 };
 
+template <class CallbackT>
+void ConcordClient::subscribe(const SubscribeRequest& request,
+                              const std::unique_ptr<opentracing::Span>& parent_span,
+                              CallbackT callback) {
+  if (not stop_subscriber_) {
```

Comment:
Let us use the classic C++ not - `!`

Reply:
Well, from my experience, this is a very rarely used feature and most C++ devs are confused when they see it.

Reply:
I don't mind changing it but we need a good reason now and for later (we can put it in our CONTRIBUTING md). I'm not sure if confused devs really counts because it won't be constant confusion and it is still proper C++. I couldn't find any guidelines around that.

Reply:
It bothers me only because from my experience most devs are surprised to see it. If you really like it, keep it, no problem.